### DISCUSSION TITLE
Количество треков в "Мне нравится" считается не совсем верно

### DISCUSSION THREAD

**Discussion by kisoid** [Chimildic/YaMuTools#70] [Помощь с ошибкой]
Вот ссылка на мой плейлист "Мне нравится": https://music.yandex.ru/users/kipo095/playlists/3

Расширение сообщает: Треков: 3444, недоступно: 1.

На самом деле, если открыть страницу https://api.music.yandex.net/users/kipo095/playlists/3
то имеем:
`result.pager.total = 3698`

а для последнего трека:
`result.tracks.3366.originalIndex = 3697`
(нумерование id начинается с 0)

Таким образом получается, что 3698 - 3367 = 331 трек не виден через этот API
Это _приватные_ + недоступные треки (у меня есть подозрение, что треки могут быть разной степени недоступности: совсем недоступные и частично недоступные или что-то вроде того...)

API, которое используется в расширении видит приватные треки, поэтому сообщает 3444 вместо 3367
Значит, кол-во приватных: 3444 - 3367 = 77
Значит, кол-во недоступных: 331 - 77 = 254

**Итого** (правильные значения):
Всего треков: 3698
Недоступно: 255 (254 + 1)
Доступно: 3443 (3444 - 1)

---
Почему это может быть важно?
Потому что лимит в 10000 треков, который существует для плейлистов, скорее всего учитывает все треки. А значит надо видеть более точное кол-во треков, чтобы не упереться в этот лимит _незаметно_.
Проблема в том, что когда лимит достигнут, новые треки "молча" перестают в него добавляться. Т.е. ты добавляешь, и они вроде как добавляются (ни ошибок, ни предупреждений не выскакивает), а по факту плейлист просто перестаёт меняться...

**Comment by Chimildic** [comment]
Добавляемое расширением _количество треков_ соответствует значению `tracksCount`, которое приходит на веб страницу вместе с данными о треках. 

API, предложенное вами, говорит мне о `total` равному 740, при этом давая данные только для 620, а плейлист из веба вообще состоит из 701. То есть 81 доступных треков где-то испарились. Загруженных руками треков у меня нет. Без данных о треках нельзя узнать доступность. 

Есть еще лучше. Со страницы всех треков, после удаления дублей, 759 лайков (24 на подозрении по имени), пусть 735. А из запроса fav-треков где лайки и дизлайки - 736 лайков. 

Внимание вопрос, сколько у меня действительно лайков? 😂 От 701 до 759? А дубли считаются в общем количестве 10 тысяч?

В общем, по мне легче на отметке в 9 тысяч продублировать плейлист с лайками в обычный плейлист или почистить его. Не выжидая 10 тысяч.

**Reply by kisoid** [reply]
Вообще трудно что-то точно сказать, не видя сам плейлист, но...
> говорит мне о total равному 740, при этом давая данные только для 620

значит 120 треков _недоступны + приватные_. Может быть, роль в доступности играет регион, из которого идёт обращение к API

>А дубли считаются в общем количестве 10 тысяч?

да, каждый дубль - занятый слот в плейлисте

> В общем, по мне легче на отметке в 9 тысяч продублировать плейлист с лайками в обычный плейлист или почистить его. Не выжидая 10 тысяч.

Ну так-то да... навряд ли будет много народу, кто столкнётся с этой проблемой... по крайней мере, в ближайшем будущем. Не все же упоротые меломаны :)
Но тем не менее - что увидел, о том написал :)

По крайней мере, я на 100% уверен, что у меня точно не 1 недоступный трек, как показывает расширение. По ощущениям, где-то штук 300 с начала года исчезло. Только за последнюю неделю штук 50 пропало (я каждый день бэкаплю плейлист своим скриптом, поэтому это точно). Так что 255 из моего расчёта - очень похоже на правду.

**Reply by Chimildic** [reply]
Недоступные лайки в плейлисте скрываются самим яндексом. Поэтому количество меньше неизвестного фактического значения.
На странице всех треков можно увидеть много недоступных при фильтре "все треки".

**Reply by kisoid** [reply]
Скрывать в плейлистах они начали в середине августа. Видимо, народ начал напрягаться от массового выкашивания треков.
Я незадолго до этого, 16.08 сделал такой плейлист: https://music.yandex.ru/users/kipo095/playlists/1254
(в нём почти все заблокированные на тот момент)
Потом спустя пару дней захожу, а плейлист пустой!
А сейчас они снова вернули отображение недоступных (но не всех) в обычных плейлистах, но не показывают их совсем в "Мне нравится". Тот единственный трек, который у меня значится как недоступный в "Мне нравится" с точки зрения расширения - это "трек доступен только по подписке", т.е. на самом деле с ним всё хорошо, просто у меня Плюса нет сейчас.
